snapshot/snapshotter.go

```diff
@@ -296,6 +296,17 @@ type Snapshotter interface {
 	// Walk all snapshots in the snapshotter. For each snapshot in the
 	// snapshotter, the function will be called.
 	Walk(ctx context.Context, fn func(context.Context, Info) error) error
+
+	// Close releases the internal resources.
+	//
+	// Close is expected to be called on the end of the lifecycle of the snapshotter,
+	// but not mandatory.
+	//
+	// Without calling Close, a new snapshotter instance may not be
```

Comment:
Again, this is an implementation detail and doesn't belong here. Snapshotters do not know about state directories and other details about how they're implemented.